### ISSUE TITLE
Signup operation taking too long

### ISSUE THREAD

**Issue by Joseph82** [awslabs/aws-mobile-appsync-sdk-android#162]
**The signup operation takes too much time on some devices.**

I am using the `com.amazonaws:aws-android-sdk-auth-userpools:2.8.4` for authentication. On some devices I have no problem, the `signUp` takes less than 1 second. However on some other devices it takes exactly 30 seconds more (30,5 seconds). I tried to understand if it could be related with the Android version, but it doesn't seem the case

this is the method I am using: `CognitoUserPool.signUp()`

Some of the devices I tried are:
1. One Plus One (Android 6) : 1 seconds
2. Samsung S8 (Android 8) : 31 seconds
3. Motorola moto G (Android 6): 31 seconds

**Comment by palpatim** [comment]
@Joseph82

Are you able to reproduce this on the latest version (2.13.3)?

Also, I see you're opening this issue on the AppSync repo. Is that because you're only able to reproduce this issue while integrating with the AppSync client, or is it something you've observed in other clients as well. I will say we haven't observed any issues like this that are tied to specific Android or Device versions (although we have of course seen other issues that vary by API levels or supported device features), so as we investigate, a minimal reproduction case using the latest version would be super helpful. If it's not related to AppSync in particular, it would also be useful to open it on the [Android SDK issues page](https://github.com/aws-amplify/aws-sdk-ios/issues/).

**Comment by Joseph82** [comment]
Hi @palpatim Yes I reproduced it also with 2.13.3, but I did some more testing and I can probably exclude a software problem on the client library. It might be, instead, a problem with the AWS backend, due to traffic limitations: when I was indeed switching to a new network, there was no delay. It might be that after reaching the limit, the request are throttled and the client has to wait.

And yes, I opened the issue here cause I am integrating it with AppSync.

So I think this issue should be closed.
